game/scripts/vscripts/abilities/oaa_elder_dragon_form.lua

```diff
@@ -131,10 +192,58 @@ if IsServer() then
 
 --------------------------------------------------------------------------------
 
+	function modifier_dragon_knight_elder_dragon_form_oaa:OnAttackLanded( event )
+		local parent = self:GetParent()
+
+		if event.attacker == parent then
+			local spell = self:GetAbility()
+
+			if spell:GetLevel() == 4 then
+				-- no rage while broken
+				if parent:PassivesDisabled() then
+					return
+				end
+
+				local chance = spell:GetSpecialValueFor( "rage_chance" )
+
+				-- we're using the modifier's stack to store the amount of prng failures
+				-- this could be something else but since this modifier is hidden anyway ...
+				local prngMult = self:GetStackCount() + 1
+
+				-- compared prng to slightly less prng
+				if RandomFloat( 0.0, 1.0 ) <= ( spell.prngTable[chance] * prngMult ) then
+					-- reset failure count
+					self:SetStackCount( 0 )
+
+					local duration = spell:GetSpecialValueFor( "rage_duration" )
+
+					-- check if the ability is already active, and if so, grab the current
+					-- duration
+					local mod = parent:FindModifierByName( "modifier_dragon_knight_dragon_form" )
+
+					if mod then
+						duration = duration + mod:GetRemainingTime()
+					end
+
+					-- apply the edf modifiers with the new duration
+					for _, modName in pairs( self.edfMods ) do
+						parent:AddNewModifier( parent, spell, modName, { duration = duration } )
+					end
+				else
+					-- increment failure count
+					self:SetStackCount( prngMult )
+				end
+			end
+		end
+	end
+
+--------------------------------------------------------------------------------
+
 	function modifier_dragon_knight_elder_dragon_form_oaa:OnRespawn( event )
 		local parent = self:GetParent()
+		local spell = self:GetAbility()
 
-		if event.unit == parent then
+		if event.unit == parent and spell:GetLevel() >= 5 then
 			local spell = self:GetAbility()
```

Comment:
Since you added this line above, you need to remove this one. The test cases prevent you from defining the same variable twice.